src/MonitorProfiler/Communication/CommandServer.cpp

```diff
@@ -72,16 +77,18 @@ void CommandServer::ListeningThread()
             continue;
         }
 
-        IpcMessage response;
-        response.MessageType = SUCCEEDED(hr) ? MessageType::OK : MessageType::Error;
-        response.Parameters = hr;
+        for (size_t i = 0; i < sizeof(HRESULT); i++)
```

Comment:
This size of an HRESULT is always 4-bytes... could this be written out explicitly without looping?

```c++
response.Payload[0] = static_cast<BYTE>(hr & 0xFF);
response.Payload[1] = static_cast<BYTE>((hr >> 8) & 0xFF);
response.Payload[2] = static_cast<BYTE>((hr >> 16) & 0xFF);
response.Payload[3] = static_cast<BYTE>((hr >> 24) & 0xFF);
```

Maybe the ` & 0xFF` is necessary to avoid sign/extension problems?

Or is there a way to use memcpy? That might have endianness problems.